eng/build.ps1

```diff
@@ -0,0 +1,417 @@
+
+    foreach ($vsixFileName in $orderedVsixFileNames) {
+        $vsixFile = Join-Path $VSSetupDir $vsixFileName
+        $fullArg = "$baseArgs $vsixFile"
+        Write-Host "`tInstalling $vsixFileName"
+        Exec-Console $vsixExe $fullArg
+    }
+}
+
+# Ensure that procdump is available on the machine.  Returns the path to the directory that contains
+# the procdump binaries (both 32 and 64 bit)
+function Ensure-ProcDump() {
+
+    # Jenkins images default to having procdump installed in the root.  Use that if available to avoid
+    # an unnecessary download.
+    if (Test-Path "C:\SysInternals\procdump.exe") {
+        return "C:\SysInternals"
+    }
+
+    $outDir = Join-Path $ToolsDir "ProcDump"
+    $filePath = Join-Path $outDir "procdump.exe"
+    if (-not (Test-Path $filePath)) {
+        Remove-Item -Re $filePath -ErrorAction SilentlyContinue
+        Create-Directory $outDir
+        $zipFilePath = Join-Path $toolsDir "procdump.zip"
+        Invoke-WebRequest "https://download.sysinternals.com/files/Procdump.zip" -UseBasicParsing -outfile $zipFilePath | Out-Null
+        Unzip $zipFilePath $outDir
+    }
+
+    return $outDir
+}
+
+function Prepare-TempDir() {
+    Copy-Item (Join-Path $RepoRoot "src\Workspaces\CoreTestUtilities\Resources\.editorconfig") $TempDir
+    Copy-Item (Join-Path $RepoRoot "src\Workspaces\CoreTestUtilities\Resources\Directory.Build.props") $TempDir
+    Copy-Item (Join-Path $RepoRoot "src\Workspaces\CoreTestUtilities\Resources\Directory.Build.targets") $TempDir
+    Copy-Item (Join-Path $RepoRoot "src\Workspaces\CoreTestUtilities\Resources\Directory.Build.rsp") $TempDir
+    Copy-Item (Join-Path $RepoRoot "src\Workspaces\CoreTestUtilities\Resources\NuGet.Config") $TempDir
+}
+
+function List-Processes() {
+    Write-Host "Listing running build processes..."
+    Get-Process -Name "msbuild" -ErrorAction SilentlyContinue | Out-Host
+    Get-Process -Name "vbcscompiler" -ErrorAction SilentlyContinue | Out-Host
+    Get-Process -Name "dotnet" -ErrorAction SilentlyContinue | where { $_.Modules | select { $_.ModuleName -eq "VBCSCompiler.dll" } } | Out-Host
+    Get-Process -Name "devenv" -ErrorAction SilentlyContinue | Out-Host
+}
+
+try {
+    Process-Arguments
+
+    . (Join-Path $PSScriptRoot "build-utils.ps1")
+
+    if ($testVsi) {
+        $ProcessesToStopOnExit += "devenv"
+    }
+
+    Push-Location $RepoRoot
+
+    if ($ci) {
+        List-Processes
+        Prepare-TempDir
+    }
+
+    if ($bootstrap) {
+        $bootstrapDir = Make-BootstrapBuild
+    }
+
+    if ($restore -or $build -or $rebuild -or $pack -or $sign -or $publish -or $testCoreClr) {
+        BuildSolution
+    }
+    
+    if ($build -and $pack -and $official) {
+        Build-OptProfData
+    }
+
+    if ($testDesktop -or $testVsi -or $testIOperation) {
+        TestUsingOptimizedRunner
+    }
+
+    if ($launch) {
+        $devenvExe = Join-Path $env:VSINSTALLDIR 'Common7\IDE\devenv.exe'
+        &$devenvExe /rootSuffix RoslynDev
```

Comment:
> &$devenvExe /rootSuffix RoslynDev [](start = 7, length = 34)

Should use Exec-Process here as the exit code isn't validated.

Reply:
Not sure why we have it but I'll update it to use `Exec-Process`